stan/math/prim/mat/fun/gp_matern52_cov.hpp

```diff
@@ -132,14 +132,15 @@ gp_matern52_cov(const std::vector<T_x> &x, const T_s &sigma,
   double five_thirds = 5.0 / 3.0;
   double neg_root_5 = -root_5;
 
-  std::vector<T_x> x_new = divide_columns(x, length_scale);
+  std::vector<Eigen::Matrix<typename return_type<T_x, T_s, T_l>::type, -1, 1>>
+      x_new = divide_columns(x, length_scale);
```

Comment:
Could divide columns just be replaced with for loops by creating an Eigen column vector to hold length scale and then using: https://github.com/stan-dev/math/blob/develop/stan/math/prim/mat/fun/elt_divide.hpp ? Then we could get rid of divide_columns all together.

I think it makes sense for length scale to be a std::vector on input here. That's the signature for all the other stuff, so there'd have to be a copy to make the column vector.

Reply:
yeah this is fine, I'll get rid of divide columns

Reply:
@bbbales2 `elt_divide` doesn't support matrix columns divided by a vector, should I add support for it or can I just use divide columns?

Reply:
I'll go ahead and add what I need to `elt_divide` when I get a chance, probs tomorrow

Reply:
Nah, don't add it to elt_divide. If it's gonna be it's own function, then keep it separate. It's kinda weird.

Reply:
got it, thanks